src/main/java/net/minecraftforge/client/model/animation/Animation.java

```diff
@@ -0,0 +1,186 @@
+package net.minecraftforge.client.model.animation;
+
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.World;
+import net.minecraftforge.client.model.IModelState;
+import net.minecraftforge.client.model.animation.ModelBlockAnimation.MBJointInfo;
+import net.minecraftforge.common.property.IUnlistedProperty;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableTable;
+
+public class Animation
+{
+    public static final IUnlistedProperty<IModelState> AnimationProperty = new IUnlistedProperty<IModelState>()
+    {
+        public String getName() { return "forge_animation"; }
+        public boolean isValid(IModelState state) { return true; }
+        public Class<IModelState> getType() { return IModelState.class; }
+        public String valueToString(IModelState state) { return state.toString(); }
+    };
+
+    public static float getWorldTime(World world)
+    {
+        return getWorldTime(world, 0);
+    }
+
+    public static float getWorldTime(World world, float tickProgress)
+    {
+        return (world.getTotalWorldTime() + tickProgress) / 20;
+    }
+
+    private static final AnimationStateMachine missing = new AnimationStateMachine(
```

Comment:
Oh absolutely, I brought it up as a discussion point. 

It just strikes me that from a semantic sense, not all models have animations. It's not that the animation was missing. It was never intended to be there in the first place, and code corresponding to that block/tileentitty/entity doesn't do animation related state changes, and that's the reason why I suggested Optional.